runtime/j9vm/javanextvmi.cpp

```diff
@@ -396,6 +408,26 @@ virtualThreadUnmountBegin(JNIEnv *env, jobject thread)
 
 	enterVThreadTransitionCritical(currentThread, thread);
 	VM_VMHelpers::virtualThreadHideFrames(currentThread, JNI_TRUE);
+
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	j9object_t carrierThreadObject = currentThread->carrierThreadObject;
+	threadObj = J9_JNI_UNWRAP_REFERENCE(thread);
+	/* Virtual thread is being umounted. If its carrier thread is suspended, spin until
+	 * the carrier thread is resumed. The carrier thread should not be mounted until it
+	 * is resumed.
+	 */
+	while (0 != J9OBJECT_U32_LOAD(currentThread, carrierThreadObject, vm->isSuspendedInternalOffset)) {
+		exitVThreadTransitionCritical(currentThread, threadObj);
+		vmFuncs->internalReleaseVMAccess(currentThread);
+		/* Spin is used instead of the halt flag; otherwise, the virtual thread will
+		 * show as suspended.
+		 */
+		f_threadSleep(10);
```

Comment:
We should make the sleep dynamically grow to a bounded maximum. This can be a TODO

Reply:
Added the TODO comments.